### PR TITLE
MahaService Post Processors

### PR DESCRIPTION
**Pull request by pranavbhole**

### CODE REVIEW

File: service/src/main/scala/com/yahoo/maha/service/MahaServiceMonitor.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Created by pranavbhole on 15/03/18.
+ */
+
+/*
+   MahaServiceMonitor is application monitoring interface for logging the api request into monitoring system
+ */
+trait MahaServiceMonitor {
+  def start: (ReportingRequest) => Unit
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/MahaServiceMonitor.scala]: I assume you meant this...
```
  def start(reportingRequest: ReportingRequest): Unit
```

File: service/src/main/scala/com/yahoo/maha/service/MahaServiceMonitor.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def stop : () => Unit
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/MahaServiceMonitor.scala]: Same here, did you mean this:
```
  def stop() : Unit
```

File: service/src/main/scala/com/yahoo/maha/service/MahaRequestProcessor.scala [hunk 3]
```diff
@@ -3,26 +3,47 @@
...
+
+trait BaseMahaRequestProcessor {
+  def process(bucketParams: BucketParams,
+              reportingRequest: ReportingRequest,
+              rawJson: Array[Byte]): MahaRequestProto.Builder
+  def onSuccess(fn: (RequestModel, RequestResult) => Unit)
+  def onFailure(fn: (GeneralError) => Unit)
+
+  //Optional Post Operation Functional Traits
+  def withRequestModelPostProcessor(fn: (Try[RequestModelResult], MahaRequestLogHelper) => Unit)
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/MahaRequestProcessor.scala]: The function doesn't produce anything?  Is it just doing validity checks?

File: service/src/main/scala/com/yahoo/maha/service/MahaRequestProcessor.scala [hunk 4]
```diff
@@ -3,26 +3,47 @@
...
+  def withRequestResultPostProcessor(fn: (Try[RequestResult], MahaRequestLogHelper) => Unit)
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/MahaRequestProcessor.scala]: Same here, it doesn't produce anything ?

File: service/src/main/scala/com/yahoo/maha/service/MahaServiceMonitor.scala (hunk 1 above)
Comment (pranavbhole) [reply service/src/main/scala/com/yahoo/maha/service/MahaServiceMonitor.scala]: yup, fixed

**Pull request closed without merging**